### PR TITLE
#167727746 user edit request

### PR DESCRIPTION
**Pull request by kola-1**
#### What this PR does
This PR will let a manager accept a trip request

#### Tasks completed
- add subrequest schema for the subrequest
- add request route to handle the request
- add subrequest route to handle subrequest
- add middleware to verify request editing authorization
- add updateTripRequest controller to update request

#### How this can be manually tested
- Clone this branch
- Change directory into it
- Run `npm install` in your terminal to install all the dependencies
- Add all the environment variables as specified in the `.env.config` file
- Run `npm run seed` in your terminal to seed the database
- Run `npm run start:dev` in your terminal to start the development server
- Test with postman or go to this URL to test with swagger `http://localhost:3000/api/v1/docs`
- Sign in as a user using this details `email: brucewayne@gmail.com, password: brucewayne`
- Send a put request with your request to be updated to `localhost:3000/api/v1/requests/edit/0fcda9b7-e17f-4162-88ed-ded031063e53` to test the request endpoint
- Send a put request with your subrequest to be updated to `localhost:3000/api/v1/subrequests/edit/01282bdb-e23f-4ece-ab08-49a1f618e66a` to test the subrequest endpoint

#### What are the relevant pivotal tracker stories?
[ #167727746](https://www.pivotaltracker.com/story/show/167727746)

### CODE REVIEW

File: src/controllers/requestController.js [hunk 1]
```diff
@@ -163,9 +163,30 @@ const updateApprovalStatus = async (req, res) => {
...
+/**
+ * update request controller
+ * @param {Object} req - server request
+ * @param {Object} res - server response
+ * @returns {Object} - custom response
+*/
+const updateTripRequest = async (req, res) => {
+  try {
+    const { requestId } = req.params;
+    const { requestType } = req;
```
Comment (jsbuddy) [inline src/controllers/requestController.js]: Line 174 and 175 can be reduced to one line. Something like this `const { requestType, params: { requestId } } = req`

File: src/controllers/requestController.js [hunk 2]
```diff
@@ -163,9 +163,30 @@ const updateApprovalStatus = async (req, res) => {
...
+    const model = (requestType === 'request') ? Request : Subrequest;
+    const options = { where: { id: requestId }, returning: true };
+    const updatedRequest = await update(model, req.body, options);
```
Comment (jsbuddy) [inline src/controllers/requestController.js]: Consider destructuring `body` and rename it to `data` from the `req` object on line 175.

File: src/middlewares/requestMiddlewares.js [hunk 3]
```diff
@@ -29,4 +29,33 @@ const verifyRequestLineManager = async (req, res, next) => {
   }
 };
 
-export default verifyRequestLineManager;
+const verifyEditRequestAuthorization = async (req, res, next) => {
+  try {
+    const { id: loggedInUserId } = req.decoded;
+    const { requestId } = req.params;
```
Comment (jsbuddy) [inline src/middlewares/requestMiddlewares.js]: Line 34 and 35 can be reduced to one line, check previous comment above.

**Comment by tejiri4** [comment]
Try your PR on heroku using the review app and add a comment if your endpoint is working on heroku

**Review by jsbuddy — CHANGES_REQUESTED** [review]

File: src/controllers/requestController.js (hunk 1 above)
Comment (kola-1) [reply src/controllers/requestController.js]: Done, thanks

File: src/routes/api/request.js [hunk 4]
```diff
@@ -500,6 +510,197 @@ const searchRequestRoute = (router) => {
...
+   *         description: Internal Server error
+   *         content:
+   *          application/json:
+   *            schema:
+   *              $ref: '#/components/schemas/ErrorResponse'
+   *     security:
+   *       - bearerAuth: []
+   */
+    .put([
+      checkToken,
```
Comment (tejiri4) [inline src/routes/api/request.js:695]: why is backlisted middleware not here?

File: src/routes/api/request.js [hunk 5]
```diff
@@ -500,6 +510,197 @@ const searchRequestRoute = (router) => {
...
+      validate(requestIdSchema),
```
Comment (tejiri4) [inline src/routes/api/request.js:605]: why is blacklisted middleware not here

**Review by tejiri4 — APPROVED** [review]
LGTM

**Pull request merged**